### PR TITLE
Changed Autostart to startmode

### PR DESCRIPTION
**Pull request by HaseenaSainul**

### CODE REVIEW

**Comment by sebaszm** [comment]
The field is still called startMode in IController. Perhaps it would be better if it's aligned to "startup" (or "startmode") everywhere?

**Comment by pwielders** [comment]
@HaseenaSainul can we aignthe field to "startmode" ?

**Comment by HaseenaSainul** [comment]
> @HaseenaSainul can we aignthe field to "startmode" ?

sure will do that

File: Source/plugins/IController.h [hunk 1]
```diff
@@ -215,7 +215,6 @@ namespace Controller {
                     RESUMED /* @text resumed */
                 };
 
-                bool AutoStart /* @brief Determines if the plugin is to be started automatically along with the framework */;
```
Comment (pwielders) [inline Source/plugins/IController.h:218]: @HaseenaSainul / @sebaszm agree it was not partof the old-code and it is not part of the PR we are resolving here but to align with the disucssion that @HaseenaSainul and I had, it would be good to also (netxt to the "startmode" have the "resumed" flag inhere as these two values together define the default startup behavior of a Service. So maybe good to add this falg (boolean) here as well ?

**Review by pwielders — CHANGES_REQUESTED** [review]
@HaseenaSainul thank you for also updating the documentation!

File: Source/plugins/IController.h (hunk 1 above)
Comment (HaseenaSainul) [reply Source/plugins/IController.h:218]: done

**Review by pwielders — DISMISSED** [review]

**Comment by sebaszm** [comment]
"resume" or "resumed"? All others are "resume**d**" :)

**Comment by HaseenaSainul** [comment]
> "resume" or "resumed"? All others are "resume**d**" :)

sorry, typo mistake. updated

**Review by sebaszm — APPROVED** [review]
